sources/System/Bonus/PlayerBonusSystem.cpp

```diff
@@ -17,9 +21,20 @@ namespace BBM
 		auto &playerBonus = entity.get<PlayerBonusComponent>();
 
 		playerBonus.nextNoClipRate -= dtime;
-		if (playerBonus.nextNoClipRate <= 0ns) {
+		if (playerBonus.nextNoClipRate <= 0ns && playerBonus.isNoClipOn) {
 			playerBonus.nextNoClipRate = playerBonus.noClipBonusRate;
 			playerBonus.isNoClipOn = false;
+			auto playerPos = entity->tryGetComponent<PositionComponent>();
+			auto playerHealth = entity->tryGetComponent<HealthComponent>();
+			if (!playerHealth || !playerPos)
+				return;
+			for (auto &[other, pos, _] : this->_wal.getScene()->view<PositionComponent, CollisionComponent>()) {
+				if (other.hasComponent<TagComponent<Player>>())
+					continue;
+				auto vec = playerPos->position.abs() - pos.position.abs();
+				if (vec.abs().x < 0.65   && vec.abs().z < 0.65 && playerPos->position.distance(pos.position) < 1)
```

Comment:
Why are you using .distance and comparing vec values?

Reply:
Because, wall are not sphere but cube so the radius won't work